src/main/java/server/ConnectionHandler.java

```diff
@@ -1,27 +1,35 @@
 package server;
 
 import server.request.Request;
-import server.request.RequestParser;
-import server.response.ResponseFormatter;
+import server.request.Parser;
+import server.response.Formatter;
 import server.response.Response;
-import server.response.ResponseSelector;
+import server.response.Selector;
+
+import java.io.IOException;
 
 public class ConnectionHandler implements Runnable {
     private Client client;
-    private RequestParser parser;
-    private ResponseSelector selector;
-    private ResponseFormatter formatter;
+    private Parser parser;
+    private Selector selector;
+    private Formatter formatter;
 
-    public ConnectionHandler(Client client, RequestParser parser, ResponseSelector selector, ResponseFormatter formatter) {
+    public ConnectionHandler(Client client, Parser parser, Selector selector, Formatter formatter) {
         this.client = client;
         this.parser = parser;
         this.selector = selector;
         this.formatter = formatter;
     }
 
     public void run() {
-        Request request = parser.parse();
-        Response response = selector.selectResponse(request);
+        Request request;
+        Response response;
+        try {
+            request = parser.parse();
+            response = selector.selectResponse(request);
+        } catch (IOException err) {
```

Comment:
While it's nice that this results in an internal server error, it's probably worth still logging this exception somewhere for debugging purposes. In a more realistic example, you'd definitely want to know what's going on internally if your service started spitting out tons of internal server errors.